### PR TITLE
Remove yum warnings while removing the Agent package.

### PR DESCRIPTION
**Pull request by remeh**
### What does this PR do?

Since 6.12.0, the `prerm` and `preinst` scripts take care of removing the embedded integrations, however, most of those files are installed by yum with the package.
A package manager tries to remove all the files that it has installed while installing a pkg, `apt` doesn't seem to complain when some of these files are missing, but `yum`/`rpm` does, and, as we're removing files on our own in scripts, it results with yum throwing a lot of warnings.

In order to avoid this, I've added a step in the `finalize` step of the package creation: it creates a file listing every files in `/opt/datadog-agent/embedded/lib/python2.7/site-packages`. This file is later used to compare what has been installed by the package and what is actually on the filesystem, to remove only what was not installed by the package.

### Motivation

Remove warnings thrown while uninstalling an RPM of the Datadog Agent, starting from RPM of the release `6.14.0`.

### Additionnal notes

Please squash before merging.

### CODE REVIEW

**Review by KSerrania — COMMENTED** [review]
A few nits on some comments to make them more accurate

File: omnibus/package-scripts/agent/prerm [hunk 1]
```diff
@@ -80,25 +80,35 @@ remove_py_compiled_files()
...
+    # Since 6.14.0, a file containing all integrations files which have been installed by
+    # the package is available. We use it to remove only the files which have *NOT*
+    # been installed by the package.
+
+    if [ -f "$INSTALL_DIR/embedded/.installed_by_pkg.txt" ]; then
+        # List all files available on the filesystem
+        find $INSTALL_DIR/embedded/lib/python*/site-packages > $INSTALL_DIR/embedded/.all-integrations.txt
+
+        # List all files available on the filesystem which were not installed by
+        # the package and rm them.
```
[the remark this answers is not in the document]
Comment (remeh) [reply omnibus/package-scripts/agent/prerm]: 😂
I'll precise this 👍

**Comment by albertvaka** [comment]
Will this work across upgrades? I'm afraid this might happen:
1. The agent is installed, finalize script creates a list of files.
2. Some integrations are installed manually.
3. The agent is upgraded, finalize script creates a list of files that now also includes manually installed integrations.

**Comment by KSerrania** [comment]
Are you talking about the `.installed_by_pkg.txt` file? If that's the case, this file is created by omnibus when we build the package, so it should never contain manually installed integrations

**Comment by albertvaka** [comment]
D'oh! I thought it was created at install time... my bad.

Then this LGTM.

**Review by albertvaka — APPROVED** [review]

**Review by olivielpeau — COMMENTED** [review]
@KSerrania @remeh I believe this does not comply with one requirement that we want to ensure: we want that the files in `site-packages` that are not related at all with official checks (for instance: custom dependencies that users may install) are _not_ removed when the Agent is upgraded.

We can't break that requirement without prior discussion, so let's revert this for now.

If we do want to break that requirement, I believe we could achieve the same result with a simpler removal logic in `postrm`, but that can be discussed later.

**Pull request merged**